tests/lib/ocpp/v201/test_smart_charging_handler.cpp

```diff
@@ -341,18 +364,127 @@ TEST_F(ChargepointTestFixtureV201, K01FR31_IfStartPeriodOfFirstChargingScheduleP
 TEST_F(ChargepointTestFixtureV201, K01FR35_IfChargingSchedulePeriodsAreNotInChonologicalOrder_ThenProfileIsInvalid) {
     auto periods = create_charging_schedule_periods({0, 2, 1});
     auto profile = create_charging_profile(DEFAULT_PROFILE_ID, ChargingProfilePurposeEnum::TxProfile,
-                                           create_charge_schedule(ChargingRateUnitEnum::A, periods), uuid());
+                                           create_charge_schedule(ChargingRateUnitEnum::A, periods), DEFAULT_TX_ID);
 
     auto sut = handler.validate_profile_schedules(profile);
 
     EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::ChargingSchedulePeriodsOutOfOrder));
 }
 
+TEST_F(ChargepointTestFixtureV201, K01_ValidateChargingStationMaxProfile_NotChargingStationMaxProfile_Invalid) {
+    create_evse_with_id(STATION_WIDE_ID);
+    auto periods = create_charging_schedule_periods({0, 1, 2});
+    auto profile = create_charging_profile(DEFAULT_PROFILE_ID + 1, ChargingProfilePurposeEnum::TxDefaultProfile,
+                                           create_charge_schedule(ChargingRateUnitEnum::A));
+
+    auto sut = handler.validate_charging_station_max_profile(profile, STATION_WIDE_ID);
+
+    EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::InvalidProfileType));
+}
+
+TEST_F(ChargepointTestFixtureV201, K04FR03_ValidateChargingStationMaxProfile_EvseIDgt0_Invalid) {
+    const int EVSE_ID_1 = DEFAULT_EVSE_ID;
```

Comment:
why do you need this? the `DEFAULT_PROFILE_ID. = 1` which is the same as the `DEFAULT_EVSE_ID`. So just validating against `DEFAULT_PROFILE_ID` should give the same result and potentially be less confusing because we validate against the same ID as the profile

Reply:
This is just an explicit name for this test (testing whether or not the EVSE ID is greater than zero) in lieu of a comment.

Reply:
My comment is not about the test name but the variable `const int EVSE_ID_1`

Reply:
Yes, that's what I meant. This is just an explicit name for the variable that's relevant to this test.